### DISCUSSION TITLE
What data is sent to chatGPT via the assistant

### DISCUSSION THREAD

**Discussion by chrleon** [zed-industries/zed#7766] [Feature Requests]
If I have repos that are forbidden to share with LLMs for company policy, can I still use Zed technically as long as I don't invoke the assistant with code from those repos? Or will the context of the open file be sent along with assistant requests?

Thanks.

**Comment by atridadl** [comment]
I would assume nothing. My basis for this:

1. Unlike Copilot Chat in VSCode, assistant does not seem to automatically build context with your open files or selected code, but instead relies on you manually adding code to the context
3. You need to provide an OpenAI key to run it, and the editor works without needing that (only assistant features wont work)

**Reply by chrleon** [reply]
Thanks @atridadl. I don't feel convinced, though. I'll wait and just keep my repos in VS Code until, and use ZED for my own private things.

**Comment by mrnugget** [comment]
Hey! Short version first: without any user interaction, nothing is shared with any LLMs. So you're good to go.

Longer version:

1. The assistant panel (in which you can chat with an OpenAI LLM) requires a custom API token and if that isn't configured, then nothing is sent to OpenAI. If it is configured and as of right now nothing is sent automatically to OpenAI either. Only the things you paste in.
2. When you enable semantic search (this requires explicit opt-in and a pop-up shows up to confirm), your codebase is sent to OpenAI to index it for semantic search.
3. When you use inline assist inside a file, then that file is sent to the LLM. If you don't use inline assist, nothing is sent to OpenAI.

**Reply by zarifpour** [reply]
How are `.env` and other files in `.gitignore` handled?

When you say "enable semantic search" are you referring to the button below:

![new items@2x](https://github.com/zed-industries/zed/assets/16494335/41770026-6e08-4439-b86f-f4127c86e2bc)

**Reply by mrnugget** [reply]
`.env`/`.gitignore` are handled like other files.

With semantic search I mean this:
![screenshot-2024-02-22-09 51 58@2x](https://github.com/zed-industries/zed/assets/1185253/0c139bf2-244a-4333-9f32-ae6ac1f3a630)

But that button that you highlighted does the same thing and when you click it, you get a prompt before anything happens:

![screenshot-2024-02-22-09 53 07@2x](https://github.com/zed-industries/zed/assets/1185253/f9094882-7a7d-4ceb-84d8-4edf0f77cc2d)

**Reply by zarifpour** [reply]
Does that mean if either of those options are enabled, the contents will be sent over to the said provider; e.g., if I opt-in the contents of the .env and other files will be processed by OpenAi?

**Reply by mrnugget** [reply]
`.gitignored` files will not be indexed: https://github.com/zed-industries/zed/blob/a194d6403c38ba2d98760bf86ebdc270ed581393/crates/semantic_index/src/semantic_index.rs#L578-L581
